File: 1829-1ForcaBruta-A.py
# https://www.urionlinejudge.com.br/judge/pt/problems/view/1829
# **************************************************************************
# *  By Elder Sobrinho - UFTM                                              *
# *  Case Input:                                                           *
# *              10                                                        *
# *              9344^9910                                                 *
# *              6575!                                                     *
# *              3537^2237                                                 *
# *              7428!                                                     *
# *              1329^1000                                                 *
# *              5066!                                                     *
# *              1707^8066                                                 *
# *              4220!                                                     *
# *              9407^2487                                                 *
# *              6572!                                                     *
# *              4553^507                                                  *
# *              9384!                                                     *
# *              9082^1917                                                 *
# *              3877!                                                     *
# *              9377^1492                                                 *
# *              8425!                                                     *
# *              2159^8895                                                 *
# *              4089!                                                     *
# *              3680^6548                                                 *
# *              9083!                                                     *
# *      Output:                                                           *
# *              Traceback (most recent call last):                        *
# *                File "teste.py", line 42, in <module>                   *
# *                  RespFac=factorial(Fact)                               *
# *                File "teste.py", line 33, in factorial                  *
# *                  return factorial(n-1)*n                               *
# *                File "teste.py", line 33, in factorial                  *
# *                  return factorial(n-1)*n                               *
# *                File "teste.py", line 33, in factorial                  *
# *                  return factorial(n-1)*n                               *
# *                [Previous line repeated 994 more times]                 *
# *                File "teste.py", line 32, in factorial                  *
# *                  if (n==1) or (n==0): return 1                         *
# *              RecursionError: maximum recursion depth exceeded in comparison*
# *     Elapsed: 0.05s                                                     *
# *  URI Output: Runtime error                                             *
# **************************************************************************
# Comentar sobre isso
#from sys import setrecursionlimit, getrecursionlimit
#setrecursionlimit(10**6)
#print(getrecursionlimit())
def factorial(n):
    if (n==1) or (n==0): return 1
    return factorial(n-1)*n
def Powx(base, exp):
    return base ** exp

n=int(input())
Pedro=Lucas=0; Out=[0]*n
for ii in range(n):
    Base,Exp=map(int,input().split("^"))
    Fact=int(input().split("!")[0])
    RespFac=factorial(Fact)
    RespPow=Powx(Base, Exp)
    if (RespFac>RespPow): Pedro+=1
    else: Lucas+=1
    Out[ii]=RespFac>RespPow
    #print(RespFac)
    #print(RespPow)

if (Pedro==Lucas): 
    print("A competicao terminou empatada!");
elif (Pedro>Lucas):
    print("Campeao: Pedro!");
else:
    print("Campeao: Lucas!");
for ii in range(n):
    if (Out[ii]):
        print("Rodada #{}: Pedro foi o vencedor".format(ii+1));
    else:
        print("Rodada #{}: Lucas foi o vencedor".format(ii+1));

File: 1829-1ForcaBruta-A.c
// https://www.urionlinejudge.com.br/judge/pt/problems/view/1829
// **************************************************************************
// *  By Elder Sobrinho - UFTM                                              *
// *  Case Input:                                                           *
// *              10                                                        *
// *              9344^9910                                                 *
// *              6575!                                                     *
// *              3537^2237                                                 *
// *              7428!                                                     *
// *              1329^1000                                                 *
// *              5066!                                                     *
// *              1707^8066                                                 *
// *              4220!                                                     *
// *              9407^2487                                                 *
// *              6572!                                                     *
// *              4553^507                                                  *
// *              9384!                                                     *
// *              9082^1917                                                 *
// *              3877!                                                     *
// *              9377^1492                                                 *
// *              8425!                                                     *
// *              2159^8895                                                 *
// *              4089!                                                     *
// *              3680^6548                                                 *
// *              9083!                                                     *
// *      Output:                                                           *
// *              Campeao: Lucas!                                           *
// *              Rodada #1: Lucas foi o vencedor                           *
// *              Rodada #2: Lucas foi o vencedor                           *
// *              Rodada #3: Lucas foi o vencedor                           *
// *              Rodada #4: Lucas foi o vencedor                           *
// *              Rodada #5: Lucas foi o vencedor                           *
// *              Rodada #6: Lucas foi o vencedor                           *
// *              Rodada #7: Lucas foi o vencedor                           *
// *              Rodada #8: Lucas foi o vencedor                           *
// *              Rodada #9: Lucas foi o vencedor                           *
// *              Rodada #10: Lucas foi o vencedor                          *
// *     Elapsed: 0.02s                                                     *
// *  URI Output: Wrong answer (10%) (Why? overflow)                        *
// **************************************************************************
#include<stdio.h>
#include<stdlib.h>
#include<math.h>
#define printf __mingw_printf // apenas se tiver usando CodeBlocks no windows. Remover no URI.

long double factorial(long double n) {
    //printf("%Lg \n", n);
    if ((n==1) || (n==0)) return 1;
   return factorial(n-1)*n;
}
long double Powx(long double base, long double exp){
    return pow(base, exp);
}
int main(){
    //printf("%5.50Lg\n", factorial(100) );
    //return;
    int n, ii, Pedro=0, Lucas=0;
    scanf("%d", &n);
    int Out[n];
    for(ii=0;   ii<n; ii++){
        int Base=57, Exp=199, Fact=874;
        scanf("%d^%d", &Base, &Exp);
        scanf("%d!", &Fact);
        long double RespFacSz=factorial(Fact);
        long double RespPowSz=Powx(Base, Exp);
        if (RespFacSz>RespPowSz) Pedro++;
        else Lucas++;
        Out[ii]=RespFacSz>RespPowSz;
    }
    if (Pedro==Lucas)
        printf("A competicao terminou empatada!\n");
    else if (Pedro>Lucas)
        printf("Campeao: Pedro!\n");
    else
        printf("Campeao: Lucas!\n");
    for(ii=0;   ii<n; ii++)
        if (Out[ii])
            printf("Rodada #%d: Pedro foi o vencedor\n",ii+1);
        else
            printf("Rodada #%d: Lucas foi o vencedor\n",ii+1);
    return 0;
}


